ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -197,34 +209,40 @@ private static void validatePseudoColumnVersion(final int pseudoColumnVersionNum
     final int version;
     final boolean mustBeMaterializedForTableJoins;
     final boolean isDisallowedForInsertValues;
+    final boolean isDisallowedInScalablePushQueries;
 
     private PseudoColumn(
         final ColumnName name,
         final SqlType type,
         final int version,
         final boolean mustBeMaterializedForTableJoins,
-        final boolean isDisallowedForInsertValues
+        final boolean isDisallowedForInsertValues,
+        final boolean isDisallowedInScalablePushQueries
```

Comment:
If we're going to add this, I might add a similar thing for pull queries and then change how I asked for the logical schema to utilize this rather than pin pull queries to an older version of the set of pseudo columns,

Reply:
What do you think is the best way of doing this?

Also, to sanity check, we'll likely remove the disallowing of these pseudocolumns pretty soon, once we start using a KafkaConsumer instead of the piggybacked processor, right? There might not be a conflict with newer pseudocolumn versions- of course, if plans change or something unforseen comes up, having a more flexible solution that doesn't rely on eventually allowing SPQs to be compatible  with ROWPARTITION/ROWOFFSET will be better.